lib/storage/src/content_manager/errors.rs

```diff
@@ -42,16 +42,54 @@ impl From<CollectionError> for StorageError {
             CollectionError::Cancelled { description } => StorageError::ServiceError {
                 description: format!("Operation cancelled: {description}"),
             },
-            err @ CollectionError::InconsistentFailure { .. } => StorageError::ServiceError {
-                description: format!("{err}"),
-            },
+            CollectionError::InconsistentShardFailure { ref first_err, .. } => {
+                let full_description = format!("{}", &err);
+                StorageError::from((*first_err.clone(), full_description))
+            }
             CollectionError::BadShardSelection { description } => {
                 StorageError::BadRequest { description }
             }
         }
     }
 }
 
+/// Used to override the `description` field of the resulting `StorageError`
+impl From<(CollectionError, String)> for StorageError {
```

Comment:
I would prefer just having a function `StorageError::from_inconsistent_shard_failure` or something like that. `From<(CollectionError, String)>` looks too generic and you have to check the comment at its exact place to check what it does.

Reply:
This is not a strong opinion, if you feel like it ok like that. I am not against merging.

Reply:
I agree with you, there is no need for a generic piece of infra here, this is a very specific case :+1: